saleor/order/views.py

```diff
@@ -124,39 +123,51 @@ def cancel_payment(request, order):
     return HttpResponseForbidden()
 
 
-def create_password(request, token):
-    if request.user.is_authenticated:
-        return redirect('order:details', token=token)
+def checkout_success(request, token):
+    """Redirects user to checkout success page or, in case of successful
+    registration, to order details page with attaching order to an account."""
     order = get_object_or_404(Order, token=token)
     email = order.user_email
+    ctx = {'email': email, 'order': order}
+    if request.user.is_authenticated:
+        return TemplateResponse(request, 'order/checkout_success.html', ctx)
     form_data = request.POST.copy()
     if form_data:
         form_data.update({'email': email})
     register_form = PasswordForm(form_data or None)
-    if User.objects.filter(email=email).exists():
-        login_form = LoginForm(initial={'login': email})
-    else:
-        login_form = None
     if register_form.is_valid():
         register_form.save()
         password = register_form.cleaned_data.get('password')
-        user = auth.authenticate(request=request, email=email,
-                                 password=password)
+        user = auth.authenticate(
+            request=request, email=email, password=password)
         auth.login(request, user)
         attach_order_to_user(order, user)
         return redirect('order:details', token=token)
-    ctx = {'form': register_form, 'email': email, 'order': order,
-           'login_form': login_form}
-    return TemplateResponse(request, 'order/create_password.html', ctx)
+    user_exists = User.objects.filter(email=email).exists()
+    login_form = LoginForm(
+        initial={'username': email}) if user_exists else None
+    ctx.update({'form': register_form, 'login_form': login_form})
+    return TemplateResponse(
+        request, 'order/checkout_success_anonymous.html', ctx)
 
 
 @login_required
 def connect_order_with_user(request, token):
-    order = get_object_or_404(
-        Order.objects.filter(user_email=request.user.email, token=token))
+    """Connects newly created order to authenticated user."""
+    try:
+        order = Order.objects.get(user_email=request.user.email, token=token)
+    except Order.DoesNotExist:
+        order = None
+    if not order:
+        msg = pgettext_lazy(
+            'Connect order with user warning message',
+            'We cannot connect order with this account due to difference '
```

Comment:
`"We couldn't assign the order to your account as the email addresses don't match"`? Please note the double quotes as there's an apostrophe in text.